### PR TITLE
chore(ports): removed unused methods

### PR DESCRIPTION
**Pull request by jack-w-shaw**
Remove some unused methods from the PortRanges interfaces. They are unused apart from tests.

Re-write tests slightly to maintain their integrity.

These interfaces are being entirely re-written to be backed by DQLite imminently. So cleaning up the interfaces here isn't strictly required.

However the fact I'm able to remove these unused functions, particularly `UniquePortRanges` and `Changes` on the `ApplicationPortRanges` and `MachinePortRanges`, confirms that we do not need to worry about them going forward into DQLite, making the work in implementing (+ reviewing) the new state and service layer easier to do

## Checklist

- [x] Code style: imports ordered, good names, simple structure, etc
- [x] Comments saying why design decisions were made
- [x] Go unit tests, with comments saying what you're testing

## QA steps

Juju compiles + Unit tests pass

### CODE REVIEW

File: state/application_test.go [hunk 1]
```diff
@@ -2518,37 +2514,25 @@ func (s *ApplicationSuite) TestDestroyWithRemovableApplicationOpenedPortRanges(c
...
 
 	err = unit1.EnsureDead()
 	c.Assert(err, jc.ErrorIsNil)
 	err = unit1.Remove(state.NewObjectStore(c, s.State.ModelUUID()))
 	c.Assert(err, jc.ErrorIsNil)
 
-	appPortRanges, err = app.OpenedPortRanges()
-	c.Assert(err, jc.ErrorIsNil)
-	c.Assert(appPortRanges.UniquePortRanges(), gc.HasLen, 2)
-
```
Comment (jack-w-shaw) [inline state/application_test.go:2551]: Before each of these removed assertions we already make an equivalent assertion on the unit.

With one exception. In this case, write a unit-based assertion instead

File: state/machine_ports_test.go [hunk 2]
```diff
@@ -340,7 +342,7 @@ func (s *MachinePortsDocSuite) TestComposedOpenCloseOperation(c *gc.C) {
 
 	// Enumerate ports
 	assertRefreshMachinePortsDoc(c, s.machPortRanges, nil)
-	unitRanges := s.machPortRanges.ForUnit(s.unit1.Name()).ForEndpoint(allEndpoints)
+	unitRanges := s.machPortRanges.ForUnit(s.unit1.Name()).UniquePortRanges()
```
Comment (jack-w-shaw) [inline state/machine_ports_test.go:345]: `ForEndpoint(allEndpoints)` is functionally the same as `UniquePortRanges`

File: state/unit_test.go [hunk 3]
```diff
@@ -1906,7 +1904,6 @@ func (s *UnitSuite) TestRemoveUnitRemovesItsPortsOnly(c *gc.C) {
 	// Verify only otherUnit still has open ports.
 	machPortRanges, err = machine.OpenedPortRanges()
 	c.Assert(err, jc.ErrorIsNil)
-	c.Assert(machPortRanges.UniquePortRanges(), gc.HasLen, 1)
 	c.Assert(machPortRanges.ForUnit(s.unit.Name()).UniquePortRanges(), gc.HasLen, 0)
 	c.Assert(machPortRanges.ForUnit(otherUnit.Name()).UniquePortRanges(), jc.DeepEquals, []network.PortRange{
 		network.MustParsePortRange("300-400/udp"),
```
Comment (jack-w-shaw) [inline state/unit_test.go:1909]: These assertions are covered by the more specific assertions following

File: state/unit_ports.go
Comment (jack-w-shaw) [inline state/unit_ports.go:1]: Reducing these interfaces is the purpose of this PR

**Review by Aflynn50 — APPROVED** [review]
LGTM

**Pull request merged**